app/kuma-dp/pkg/dataplane/metrics/metrics_producer.go

```diff
@@ -85,7 +85,9 @@ func (ap *AggregatedProducer) Produce(ctx context.Context) ([]metricdata.ScopeMe
 func combineMetrics(metrics <-chan *metricdata.ScopeMetrics) []metricdata.ScopeMetrics {
 	var combinedMetrics []metricdata.ScopeMetrics
 	for metric := range metrics {
-		combinedMetrics = append(combinedMetrics, *metric)
+		if metric != nil {
```

Comment:
when can this be nil?

Reply:
when [fetchStats](https://github.com/kumahq/kuma/pull/9424/files#diff-1e652cf18e10c3d480b68f787cb5dba80b4847cc44ea6b53857f24e6c7aeef0cL93) failed, and we could not get metrics or map them. I stumbled upon this while testing this with an application that returned metrics in wrong format